src/Atlas2DSlice.ts

```diff
@@ -147,15 +169,11 @@ export default class Atlas2DSlice implements VolumeRenderImpl {
 
       this.setUniform("AABB_CLIP_MIN", bounds.bmin);
       this.setUniform("AABB_CLIP_MAX", bounds.bmax);
-      const slice = Math.floor(this.settings.zSlice);
-      const sizez = this.volume.imageInfo.volumeSize.z;
-      if (slice >= 0 && slice <= sizez - 1) {
-        this.setUniform("Z_SLICE", slice);
-        const sliceRatio = slice / sizez;
-        this.volume.updateRequiredData({
-          subregion: new Box3(new Vector3(0, 0, sliceRatio), new Vector3(1, 1, sliceRatio)),
-        });
-      }
+      this.updateSlice();
+      const sliceRatio = Math.floor(this.settings.zSlice) / this.volume.imageInfo.volumeSize.z;
+      this.volume.updateRequiredData({
+        subregion: new Box3(new Vector3(0, 0, sliceRatio), new Vector3(1, 1, sliceRatio)),
+      });
```

Comment:
+1 for Peyton's comment. The out of bounds check probably should skip this code too, as it was before.  I imagine that the upper levels of the code really prevent it from being out of range but it's still a good sanity check here. Does that sound right?

Reply:
Yeah, I'd bet this can never occur in practice, but it's still a good idea to preserve the sanity check.